Resources/Catalog/RucioFileCatalogClient.py

```diff
@@ -0,0 +1,545 @@
+from DIRAC.Core.Utilities.List                          import breakListIntoChunks
+from DIRAC.Core.Security.ProxyInfo                      import getProxyInfo, formatProxyInfoAsString
+from DIRAC.ConfigurationSystem.Client.Helpers.Registry  import getDNForUsername, getVOMSAttributeForGroup, \
+                                                               getVOForGroup, getVOOption
+from DIRAC.Resources.Catalog.FileCatalogClientBase import FileCatalogClientBase
+
+#from DIRAC.Core.Base import Script
+#Script.parseCommandLine()
+
+from rucio.client import Client
+from rucio.common.exception import DataIdentifierNotFound, DuplicateContent, FileReplicaAlreadyExists, \
+                                   FileAlreadyExists, CannotAuthenticate, MissingClientParameter
+from rucio.common.utils import chunks, extract_scope
+
+
+def get_scope(lfn, scopes=[], client=None):
+  scope, name = extract_scope(did=lfn, scopes=scopes)
+  return scope
+
+
+class RucioFileCatalogClient(FileCatalogClientBase):
+
+  READ_METHODS = FileCatalogClientBase.READ_METHODS + ['isLink', 'readLink', 'isFile', 'getFileMetadata',
+                                                       'getReplicas', 'getReplicaStatus', 'getFileSize',
+                                                       'isDirectory', 'getDirectoryReplicas',
+                                                       'listDirectory', 'getDirectoryMetadata',
+                                                       'getDirectorySize', 'getDirectoryContents',
+                                                       'resolveDataset', 'getLFNForPFN', 'getUserDirectory']
+
+
+  WRITE_METHODS = FileCatalogClientBase.WRITE_METHODS + ['createLink', 'removeLink', 'addFile', 'addReplica',
+                                                         'removeReplica', 'removeFile', 'setReplicaStatus',
+                                                         'setReplicaHost', 'createDirectory', 'removeDirectory',
+                                                         'removeDataset', 'removeFileFromDataset', 'createDataset',
+                                                         'changePathOwner', 'changePathMode']
+
+
+  NO_LFN_METHODS = FileCatalogClientBase.NO_LFN_METHODS + ['getUserDirectory', 'createUserDirectory',
+                                                           'createUserMapping', 'removeUserDirectory']
+
+
+  ADMIN_METHODS = FileCatalogClientBase.ADMIN_METHODS + ['getUserDirectory', 'createUserDirectory',
+                                                         'createUserMapping', 'removeUserDirectory']
+
+  def __init__(self, **options):
+    self.convertUnicode = True
+    proxyInfo = {'OK': False}
+    if not os.getenv('X509_USER_PROXY'):
+      proxyInfo = getProxyInfo(disableVOMS=True)
+      if proxyInfo['OK']:
+        os.environ['X509_USER_PROXY'] = proxyInfo['Value']['path']
+        gLogger.debug('X509_USER_PROXY not defined. Using %s' % proxyInfo['Value']['path'])
```

Comment:
Don't use directly `gLogger`. Instead use
```
self.log = gLogger.getSubLogger('RucioFileCatalog')
```
and then use `self.log`

Reply:
Actually, we agreed to rather do `sLog = gLogger.getSubLogger(__name__)` as a global variable and use that